src/panels/ActionsPanel/ActionsPanel.tsx

```diff
@@ -1,13 +1,124 @@
+import RadioInputBox from "components/RadioInputBox/RadioInputBox";
+
+import "./_actions-panel.scss";
+
+type ActionData = {
+  [key: string]: ActionItem;
+};
+
+type ActionItem = {
+  description: string;
+  params: ActionParams;
+};
+
+type ActionParams = {
+  description: string;
+  properties: ActionProps;
+  required: string[];
+  title: string;
+  type: string;
+};
+
+type ActionProps = {
+  [key: string]: ActionProp;
+};
+
+type ActionProp = {
+  description: string;
+  type: string;
+};
+
+export type ActionOptions = ActionOptionDetails[];
+
+type ActionOptionDetails = {
+  name: string;
+  description: string;
+  type: string;
+  required: boolean;
+};
 
 export default function ActionsPanel(): JSX.Element {
-  const title = <div>0 3 units selected</div>;
+  const appStore = useStore();
+  const appState = appStore.getState();
+  const { appName, modelName } = useParams<EntityDetailsRoute>();
+  const getModelUUIDMemo = useMemo(() => getModelUUID(modelName), [modelName]);
+  // Selectors.js is not typescript yet and it complains about the return value
+  // of getModelUUID. TSFixMe
+  const modelUUID = useSelector(
+    getModelUUIDMemo as (state: DefaultRootState) => unknown
+  );
+  const [actionData, setActionData] = useState<ActionData>();
+
+  useEffect(() => {
+    getActionsForApplication(appName, modelUUID, appStore.getState()).then(
+      (actions) => {
+        if (actions?.results?.[0]?.actions) {
+          setActionData(actions.results[0].actions);
+        }
+      }
+    );
+  }, [appName, appStore, modelUUID]);
+
+  // See above note about selectors.js typings TSFixMe
+  const namespace =
+    appState.juju?.modelData?.[modelUUID as string]?.applications?.[appName]
+      ?.charm;
+
+  const generateSelectedUnitList = () => "1, 2, 3";
+
+  const generateTitle = () => (
+    <h5>{generateIconImg(appName, namespace)} 3 units selected</h5>
+  );
+
   return (
     <Aside width="narrow">
       <div className="p-panel actions-panel">
-        <PanelHeader title={title} />
+        <PanelHeader title={generateTitle()} />
+        <div className="actions-panel__unit-list">
+          Run action on {appName}: {generateSelectedUnitList()}
+        </div>
+        <div className="actions-panel__action-list">
+          {generateActionlist(actionData)}
+        </div>
       </div>
     </Aside>
   );
 }
+
+function generateActionlist(actionData: ActionData | undefined) {
+  if (!actionData) return null;
+  return Object.keys(actionData).map((actionName) => {
+    const action = actionData[actionName];
+    const options: ActionOptions = [];
+
+    Object.keys(action.params.properties).forEach((name) => {
+      const property = action.params.properties[name];
+      options.push({
+        name: name,
+        description: property.description,
+        type: property.type,
+        required: action.params.required.includes(name),
+      });
+    });
+
+    return (
+      <RadioInputBox
```

Comment:
Maybe until we have the ability to trigger the action this should just be a basic list?

Reply:
I think there is enough complexity in the interactions right now to keep it. This is going to likely change in the next PR because of the change to `Formik`

Reply:
My point is it's confusing that you provide the field options to run an action but no button to run it.

Reply:
Sure thing, converting this to a div for this branch.